src/core.js

```diff
@@ -37,6 +39,9 @@ class Comb {
         return new minimatch.Minimatch(pattern);
       });
 
+    if (config.syntax)
+      this.syntaxMap = config.syntax;
```

Comment:
... but may be `this.syntaxMap = util.extend({}, config.syntax)` would be better here?